models/models/markov_chain/markov_chain.py

```diff
@@ -34,28 +37,31 @@ def train(self, epochs: int, x_train: Any, y_train: Any, progress_callback: Prog
         for i in range(n_gram_next.shape[0]):
             n_gram_next[i] = []
 
+        start = time.time()
+        time.perf_counter()
+
         for i in range(len(self.data)):
+
+            elapsed = time.time() - start
+            progress_callback([(elapsed, 100 * i / len(self.data))])
+
             for j in range(len(self.data[i]) - n - 1):
                 curr_n_gram = tuple(self.data[i][j:j + n])
                 next_note = self.data[i][j + n + 1]
                 n_gram_next[self.n_grams_list.index(
                     curr_n_gram)].append(next_note)
 
+        elapsed = time.time() - start
+        progress_callback([(elapsed, len(self.data))])
```

Comment:
Did you mean
```suggestion
        progress_callback([(elapsed, 100)])
```
?

Reply:
No, the callback is reporting time elapsed and number of data sample being processed. This executes after processing the last sample.

Reply:
> No, the callback is reporting time elapsed and number of data sample being processed

Are you sure? It seems to report % (https://github.com/piotrowskv/music_generation/pull/59/files#diff-6dff6b1b139706191ec2b415c052507856ff8905cab45498beba181a4a9b65e4R44) also the metadata says these are %

Reply:
Ohh, yeah you're right